packages/amplify_core/lib/src/types/notifications/push/remote_message_callback.dart

```diff
@@ -0,0 +1,8 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import '../notification_types.dart';
+
+typedef RemoteMessageCallback = void Function(
```

Comment:
We probably need to do some clean up after the callback completion, where a developer may initiate some async operations. So let's use `FutureOr` so we can `await` for before starting cleanup.

```suggestion
typedef RemoteMessageCallback = FutureOr<void> Function(
```